scripts/tank_cmd.py

```diff
@@ -1096,6 +1099,14 @@ def run_engine_cmd(log, pipeline_config_root, context_items, command, using_cwd,
             exit_code = show_help(logger)
             sys.exit(exit_code)
 
+    # If we are not authenticated.
+    if not shotgun.is_script_user_authenticated():
```

Comment:
wondering if we even need this `is_script_user_authenticated` check? could it not just simply call `console_login` and `logout` and in the case where there are ready-set credentials, it just silently does the right thing? This way, the session stuff doesn't have to expose any notion of different authentication modes?